grails-app/domain/com/unifina/domain/security/IntegrationKey.groovy

```diff
@@ -48,7 +48,11 @@ class IntegrationKey implements Serializable {
 	@CompileStatic
 	private Map jsonMap() {
 		if (service == Service.ETHEREUM || service == Service.ETHEREUM_ID) {
-			return [address: ((JSONObject) JSON.parse(json)).get("address")]
+			JSONObject jso = (JSONObject) JSON.parse(json)
+			Map jsmap = [address: jso.get("address")]
+			if(jso.has("privateKeyPlaintext"))
```

Comment:
If `service` is `ETHEREUM`, we should read and decrypt the encrypted private key from the `privateKey` field, and write it to the resulting map (perhaps as key `privateKey` instead of `privateKeyPlaintext` for brevity)